pkg/streamingpromql/functions/common.go

```diff
@@ -43,3 +46,21 @@ func FloatTransformationDropHistogramsFunc(transform func(f float64) float64) In
 func Passthrough(seriesData types.InstantVectorSeriesData, _ *pooling.LimitingPool) (types.InstantVectorSeriesData, error) {
 	return seriesData, nil
 }
+
+// RangeVectorStepFunction is a function that operates on a range vector step.
+//
+// floatBuffer and histogramBuffer will contain the values for the step range.
+// They may also contain values outside of the step, so step.RangeEnd needs to be considered.
+//
+// Parameters:
+//   - step: the range vector step data to be processed.
+//   - rangeSeconds: the duration of the range in seconds.
+//   - floatBuffer: a ring buffer containing float values for the step range.
+//   - histogramBuffer: a ring buffer containing histogram values for the step range.
+//
+// Returns:
+//   - hasFloat bool: a boolean indicating if a float value is present.
+//   - f float64: the float value.
+//   - h *histogram.FloatHistogram: nil if no histogram is present.
+//   - err error.
+type RangeVectorStepFunction func(step types.RangeVectorStepData, rangeSeconds float64, floatBuffer *types.FPointRingBuffer, histogramBuffer *types.HPointRingBuffer) (hasFloat bool, f float64, h *histogram.FloatHistogram, err error)
```

Comment:
[nit] The convention for these OK / present bools is for them to follow the parameter they relate to:
```suggestion
type RangeVectorStepFunction func(step types.RangeVectorStepData, rangeSeconds float64, floatBuffer *types.FPointRingBuffer, histogramBuffer *types.HPointRingBuffer) (f float64, hasFloat bool, h *histogram.FloatHistogram, err error)
```